src/spyglass/utils/dj_mixin.py

```diff
@@ -530,8 +565,138 @@ def delete(self, force_permission=False, *args, **kwargs):
+
+        return ExportSelection()
+
+    @property
+    def export_id(self):
+        """ID of export in progress.
+
+        NOTE: User of an env variable to store export_id may not be thread safe.
+        Exports must be run in sequence, not parallel.
+        """
+
+        return int(environ.get(EXPORT_ENV_VAR, 0))
+
+    @export_id.setter
+    def export_id(self, value):
+        """Set ID of export using `table.export_id = X` notation."""
+        if self.export_id != 0 and self.export_id != value:
+            raise RuntimeError("Export already in progress.")
+        environ[EXPORT_ENV_VAR] = str(value)
+        exit_register(self._export_id_cleanup)  # End export on exit
+
+    @export_id.deleter
+    def export_id(self):
+        """Delete ID of export using `del table.export_id` notation."""
+        self._export_id_cleanup()
+
+    def _export_id_cleanup(self):
+        """Cleanup export ID."""
+        if environ.get(EXPORT_ENV_VAR):
+            del environ[EXPORT_ENV_VAR]
+        exit_unregister(self._export_id_cleanup)  # Remove exit hook
+
+    def _start_export(self, paper_id, analysis_id):
+        """Start export process."""
+        if self.export_id:
+            logger.info(f"Export {self.export_id} in progress. Starting new.")
+            self._stop_export(warn=False)
+
+        self.export_id = self._export_table.insert1_return_pk(
+            dict(
+                paper_id=paper_id,
+                analysis_id=analysis_id,
+                spyglass_version=self._spyglass_version,
+            )
+        )
+
+    def _stop_export(self, warn=True):
+        """End export process."""
+        if not self.export_id and warn:
+            logger.warning("Export not in progress.")
+        del self.export_id
+
+    def _log_fetch(self, *args, **kwargs):
+        """Log fetch for export."""
+        if not self.export_id or self.database == "common_usage":
+            return
+        logger.debug(f"Export {self.export_id}: fetch()   {self.table_name}")
+
+        restr = self.restriction or True
+        limit = kwargs.get("limit")
+        offset = kwargs.get("offset")
+        if limit or offset:
+            restr = super().fetch(  # Use result as restr if limit/offset
+                restr, as_dict=True, limit=limit, offset=offset
+            )
+        restr_str = make_condition(self, restr, set())
+
+        if isinstance(restr_str, str) and len(restr_str) > 2048:
+            raise RuntimeError(
+                "Export cannot handle restrictions > 2048.\n\t"
+                + "If required, please open an issue on GitHub.\n\t"
+                + f"Restriction: {restr_str}"
+            )
+        self._export_table.Table.insert1(
+            dict(
+                export_id=self.export_id,
+                table_name=self.full_table_name,
+                restriction=make_condition(self, restr_str, set()),
+            ),
+            skip_duplicates=True,
+        )
```

Comment:
When you get a list of restriction keys from the fetch call above you get very long queries with multiple OR gates.  Instead add each restriction entry to the table separately.
```python
        if not isinstance(restr, List):
            restr = [restr]

        for r in restr:
            restr_str = make_condition(self, r, set())

            if isinstance(restr_str, str) and len(restr_str) > 2048:
                raise RuntimeError(
                    "Export cannot handle restrictions > 2048.\n\t"
                    + "If required, please open an issue on GitHub.\n\t"
                    + f"Restriction: {restr_str}"
                )
            self._export_table.Table.insert1(
                dict(
                    export_id=self.export_id,
                    table_name=self.full_table_name,
                    restriction=make_condition(self, restr_str, set()),
                ),
                skip_duplicates=True,
            )

```

Reply:
I ran into issues with this approach when `restr` is an `AndList` because...
```python
Table & A & B
```
would get logged as...
```python
Table & A
Table & B
```

It's also not clear to me how we benefit from separating `Or` lists because the `RestrGraph` will have to merge each of the entries later on the same node. We'd be less likely to hit the varchar limit on the Selection table, but we'd still hit it on she Export table ... if someone is using `limit` in their fetch. `limit` is also not super predictable, so it seems like very much an edge case anyway

Reply:
Might work with 
```python
if not isinstance(restr, Iterable) or isinstance(restr, dj.AndList):
    restr = [restr]
```
But I'm still dissuaded by the need to reassemble